script/test-robots

```diff
@@ -28,22 +28,26 @@ from controller_utils import NUM_ZONES  # isort:skip
 EXCESS_SUBPROCESS_TIMEOUT = 10
 
 
-def test_webots_runtime(comp_match_script: str, temp_archives: Path) -> int:
-    """
-    Run a empty webots match to determine the expected duration of a successful match.
-    temp_archives are used to recieve the logs and animations.
-    """
+def run_robot_match(
+    comp_match_script: Path,
+    archives_dir: Path,
+    tla: str,
+    failed_tlas: Optional[List[str]],
```

Comment:
Is this effectively an output parameter? I would encourage not doing this and instead having this util return a value indicating whether or not the run succeeded.

Python allow returning tuples, so this is fairly easy.

However I would also encourage not returning the duration - that (as far as I can tell) is only used for the calibration and so is a secondary consideration. Moreover it's also something which the caller can trivially measure for themselves.

Reply:
39f318d69424357ecb9c58fd9317ddd1759caac5